.github/workflows/build_and_test.yml

```diff
@@ -4650,6 +4761,42 @@ jobs:
         call setenv.cmd
         cd DevGuideExamples\DCPS\Messenger
         perl run_test.pl -ExeSubDir Static_Debug
+    - name: build messenger tests
+      shell: cmd
+      run: |
+        cd OpenDDS
+        call setenv.cmd
+        cd tests\DCPS\Messenger
+        perl %ACE_ROOT%\bin\mwc.pl -type vs2019 -static
+    - name: make messenger tests
+      shell: cmd
+      run: |
+        cd OpenDDS
+        call setenv.cmd
+        cd tests\DCPS\Messenger
+        msbuild -p:Configuration=Debug,Platform=x64 -m Messenger.sln
+    - name: build compiler tests
+      shell: cmd
+      run: |
+        cd OpenDDS
+        call setenv.cmd
+        cd tests\DCPS\Compiler
+        perl %ACE_ROOT%\bin\mwc.pl -type vs2019 -static
+    - name: make compiler tests
+      shell: cmd
+      run: |
+        cd OpenDDS
+        call setenv.cmd
+        cd tests\DCPS\Compiler
+        dir
+        msbuild -p:Configuration=Debug,Platform=x64 -m Compiler.sln
+    - name: run Static CI tests
```

Comment:
Let's keep the pattern of running the tests in the test job. And especially for the static test jobs, which may build & run separate tests, so they can reuse the core OpenDDS build artifact.